File: BlackJack.WebApi/Repository/GameRepository.cs
using BlackJack.Models;
using BlackJack.WebApi.Interface;
using BlackJack.WebApi.Models;
using Microsoft.Extensions.Configuration;
using System.Collections.Generic;

namespace BlackJack.WebApi.Repository
{
    public class GameRepository :IGame
    {
        private readonly IConfiguration _configuration;
        private readonly CardDeskRepository _cardDeskRepository;
        
        
        public GameRepository(IConfiguration configuration)
        {
            _configuration = configuration;
            _cardDeskRepository = new CardDeskRepository(configuration);
        }
        public GameCustomModel StartGame()
        {
            var playerCards = new List<Hand>();
            var dealerCards = new List<Hand>();
            var player = new CardDeck { PlayerName = "Player", CountofPlayingRounds = 0 };
            var dealer = new CardDeck { PlayerName = "Dealer", CountofPlayingRounds = 0 };

            playerCards.Add(_cardDeskRepository.DrawCard(player));
            playerCards.Add(_cardDeskRepository.DrawCard(player));
            dealerCards.Add(_cardDeskRepository.DrawCard(dealer));
            dealerCards.Add(_cardDeskRepository.DrawCard(dealer));
            var model = new GameCustomModel
            {
                Dealers = dealerCards,
                Player  = playerCards
            };

            return model;
        }
    }
}

File: BlackJack.TypeScript/src/repositories/GameRepository.ts
import { CardDeck } from "../models/CardDeck";
import { GameCustomModel } from "../models/GameCustomModel";
import { Hand } from "../models/Hand";
import { CardDeskRepository } from "./CardDeckRepository";


 class GameRepository{
 
    cardDeckRepository!: CardDeskRepository;
    player!: CardDeck;
    dealer!: CardDeck;

    constructor( ) {
       this.cardDeckRepository = new CardDeskRepository();      
       this.player = new CardDeck(0,'Player');
       this.dealer = new CardDeck(0,'Dealer')
    }

    gameCustomModel: any StartGame(): any{
       
        const playerCards: Array<Hand> = new Array();
        const dealerCards: Array<Hand> = new Array();
        
        playerCards.push(this.cardDeckRepository.drawCard(this.player));
        playerCards.push(this.cardDeckRepository.drawCard(this.player));
        dealerCards.push(this.cardDeckRepository.drawCard(dealer));
        dealerCards.push(this.cardDeckRepository.drawCard(dealer));
        var model = new GameCustomModel()
       
            model.dealer = dealerCards,
            model.player  = playerCards
        
        return model;
    }


}
export default new GameRepository();